.github/workflows/test-build.yml

```diff
@@ -0,0 +1,41 @@
+name: ci
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+concurrency:
+  # Cancel any running workflow for the same branch when new commits are pushed.
+  # We group both by ref_name (available when CI is triggered by a push to a branch/tag)
+  # and head_ref (available when CI is triggered by a PR).
+  group: "${{ github.ref_name }}-${{ github.head_ref }}"
+  cancel-in-progress: true
+
+jobs:
+  test-build:
+    runs-on: ubuntu-latest
```

Comment:
[nit] Might be nice to set a human-friendly name for the job with `name`, something like `Build and test on Go ${{ matrix.go-version }}` perhaps?

Reply:
I tried to change this, but it seems the job id strictly just the alphabet and `-` and `_` with no space. Of course no templating too apparently. So I reverted to the original ID `test-build` which clearly describes what the job do. Let me know if you have a better suggestion

Reply:
Oh now I just realized you are talking about run-name. Let try to apply that.

Reply:
Applied in ff313735d90e6298a72ffa63bc6ef246e493c81b